backend/src/webhook/elopage.ts

```diff
@@ -0,0 +1,370 @@
+
+    Additional we have the Elopage API docu:
+        https://apidoc.elopage.com/#webhooks
+
+    I assume that the webhook arrives via POST and transmits a string as shown above
+*/
+
+import { LoginElopageBuys } from '@entity/LoginElopageBuys'
+import { LoginUser } from '@entity/LoginUser'
+import { randomBytes } from 'crypto'
+import { UserResolver } from '../graphql/resolver/UserResolver'
+
+export const elopageWebhook = async (req: any, res: any): Promise<void> => {
+  res.status(200).end() // Responding is important
+
+  const loginElopgaeBuy = new LoginElopageBuys()
+  let firstName = ''
+  let lastName = ''
+  const entries = req.body.split('&')
+  entries.map((entry: string) => {
+    const keyVal = entry.split('=')
+    if (keyVal.length !== 2) {
+      throw new Error(`Error parsing entry '${entry}'`)
+    }
+    const key = keyVal[0]
+    const val = decodeURIComponent(keyVal[1]).replace('+', ' ').trim()
+    switch (key) {
+      case 'product[affiliate_program_id]':
+        loginElopgaeBuy.affiliateProgramId = parseInt(val)
+        break
+      case 'publisher[id]':
+        loginElopgaeBuy.publisherId = parseInt(val)
+        break
+      case 'order_id':
+        loginElopgaeBuy.orderId = parseInt(val)
+        break
+      case 'product_id':
+        loginElopgaeBuy.productId = parseInt(val)
+        break
+      case 'product[price]':
+        // TODO: static_cast<Poco::Int32>(round(stof(temp) * 100.0f));
+        loginElopgaeBuy.productPrice = parseFloat(val)
+        break
+      case 'payer[email]':
+        loginElopgaeBuy.payerEmail = val
+        break
+      case 'publisher[email]':
+        loginElopgaeBuy.publisherEmail = val
+        break
+      case 'payment_state':
+        loginElopgaeBuy.payed = val === 'paid'
+        break
+      case 'success_date':
+        loginElopgaeBuy.successDate = new Date(val)
+        break
+      case 'event':
+        loginElopgaeBuy.event = val
+        break
+      case 'membership[id]':
+        // TODO this was never set on login_server - its unclear if this is the correct value
+        loginElopgaeBuy.elopageUserId = parseInt(val)
+        break
+      case 'payer[first_name]':
+        firstName = val
+        break
+      case 'payer[last_name]':
+        lastName = val
+        break
+      default:
+        // eslint-disable-next-line no-console
+        console.log(`Unknown Elopage Value '${entry}'`)
+    }
+    return null // we write things into the loginElopgaeBuy object, no return value needed
```

Comment:
Please remove this line. It is confusing. We do not need a return value for the map function, as you say. As it is, it looks like the return statements in the following if clauses, which is not the same.

Reply:
A map **must** return a value

Reply:
use apply?